walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,91 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public final class FindQuery {
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
+        // Get attendees and duration
+        Collection<String> attendees = request.getAttendees();
+        long duration = request.getDuration();
+
+        // Create list to store available timings
+        Collection<TimeRange> times = new ArrayList<>();
+
+        // if there are no attendees, return the whole day
+        if (attendees.isEmpty()) {
+            times.add(TimeRange.WHOLE_DAY);
+            return times;
+        }
+
+        // if the duration is longer than a day, return no timings (empty list)
+        if (duration >= TimeRange.WHOLE_DAY.duration() + 1) {
+            return times;
+        }
+
+        // get unavailable event timings
+        List<Event> unavailableTimings = getUnavailableTimings(events, attendees);
+
+        // get available event timings
+        times = getAvailableTimings(unavailableTimings, duration);
+
+        return times;
+    }
+
+    // Get unavailable timings for events, sort events and return the list of timings
+    private List<Event> getUnavailableTimings(Collection<Event> events, Collection<String> attendees) {
+        List<Event> unavailableTimings = new ArrayList<>();
+        for (Event e : events) {
+            for (String a : attendees) {
+                if (e.getAttendees().contains(a)) {
+                    unavailableTimings.add(e);
+                    break;
+                }
+            }
+        }
+
+        Comparator<Event> compareStartingTime =
+                (Event e1, Event e2) -> TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen());
+        Collections.sort(unavailableTimings, compareStartingTime);
+
+        return unavailableTimings;
+    }
+
+    // Get TimeRange of available timings
+    private Collection<TimeRange> getAvailableTimings(List<Event> unavailableTimings, long duration) {
+        Collection<TimeRange> times = new ArrayList<>();
```

Comment:
small nit: could you use a better name than "times", "tr", "trBefore" and "trAfter"? don't worry about being verbose :)